src/bio.h

```diff
@@ -10,6 +10,26 @@
 #define __BIO_H
 
 typedef void lazy_free_fn(void *args[]);
+typedef void comp_fn(uint64_t user_data);
```

Comment:
In the future it may not always be client_id.
```suggestion
typedef void comp_fn(void *user_data);
```

Reply:
If it's okay with you and as long there are no new requirements, I'd prefer to keep it slightly more constrained and easier to read.